jenkins/release-workflows/release-notes-generate.jenkinsfile

```
@@ -0,0 +1,171 @@
+            description: 'in format yyyy-mm-dd, example 2022-07-26.',
+            trim: true
+        )
+        string(
+            name: 'COMPONENTS',
+            description: 'Optional: If this field contains one or more component names (e.g. OpenSearch common-utils ...), will build with "--component COMPONENT_NAME_HERE ...", else build everything in the INPUT_MANIFEST.',
+            trim: true
+        )
+        string(
+            name: 'MODEL_ID',
+            description: 'AWS Bedrock model id to use, defaults to Anthropic Claude 3.7v2',
+            trim: true
+        )
+        string(
+            name: 'MAX_TOKENS',
+            description: 'Max number of tokens to use for supplied model id, deafults to 2000',
+            trim: true
+        )
+        string(
+            name: 'REF',
+            defaultValue: 'main',
+            description: 'Override input component github repo ref, defaults to provided in input manifest file',
+            trim: true
+        )
+    }
+    stages {
+        stage('Parameters Check') {
+            agent { label 'Jenkins-Agent-AL2023-X64-M54xlarge-Docker-Host' }
+            steps {
+                script {
+                    if (INPUT_MANIFEST.isEmpty() || GIT_LOG_DATE.isEmpty()) {
+                        currentBuild.result = 'ABORTED'
+                        error('INPUT_MANIFEST or GIT_LOG_DATE parameters cannot be empty!')
+                    }
+                }
+            }
+        }
+        stage('Generate consolidated release notes') {
+            agent { label 'Jenkins-Agent-AL2023-X64-M54xlarge-Docker-Host' }
+            steps {
+                script {
+                    def inputManifestObj = lib.jenkins.InputManifest.new(readYaml(file: "manifests/${params.INPUT_MANIFEST}"))
+                    def componentDefaultList = inputManifestObj.getNames()
+                    def product = inputManifestObj.build.name
+                    def componentList = COMPONENTS ? COMPONENTS.trim().split(' ') as List : componentDefaultList
+                    println("Component list: ${componentList}")
+                    def componentNotes = [:]
+                    stash includes: '**', name: "release-notes-opensearch-$BUILD_NUMBER"
+
+                    // Method 1: Use eachWithIndex to capture values properly
+                    componentList.eachWithIndex { component, component_index ->
+                        def wait_seconds = component_index * 30
+                        // Capture the values in local variables for the closure
+                        def currentComponent = component
+                        def currentIndex = component_index
+                        def currentWaitSeconds = wait_seconds
+
+                        if (!(currentComponent in ['notifications-core', 'functionalTestDashboards'])) {
+                            componentNotes["Release notes for ${currentComponent}"] = {
+                                timeout(time: 4, unit: 'HOURS') {
```

4 is too much I believe
```suggestion
                                timeout(time: 1, unit: 'HOURS') {
```